### PR TITLE
Bug 1884270: bypass golang url parsing with scp styled ssh git URLs; refactor URL for older git clients

### PR DESCRIPTION
**Pull request by gabemontero**
/assign @adambkaplan

### CODE REVIEW

File: pkg/scm/git/url.go [hunk 1]
```diff
@@ -106,6 +108,14 @@ func Parse(rawurl string) (*URL, error) {
 		}, nil
 	}
 
+	// if ssh://git@github.com:sclorg/nodejs-ex then strip ssh:// a la what we
+	// see in other upstream git parsing handling of scp styled git URLs;
+	//TODO is there someway to infer git v1 vs. v2 so we do not have to strip this
+	// for older clients
```
Comment (gabemontero) [inline pkg/scm/git/url.go:114]: @adambkaplan let's discuss this in scrum if not in line here

Comment (adambkaplan) [reply pkg/scm/git/url.go]: Per discussion on slack - we shouldn't worry about git version.

**Comment by gabemontero** [comment]
OK @adambkaplan per slack discussion let's go with this .... PTAL

**Review by adambkaplan — APPROVED** [review]
/lgtm

**Pull request merged**